x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/autocomplete_field/index.tsx

```diff
@@ -38,15 +40,19 @@ export const AutocompleteField = ({
     [field]
   );
 
-  const selectedField = useMemo(() => {
-    const existingField = (field.value as string) ?? '';
-    const [newSelectedField] = indices.fields.filter(
-      ({ name }) => existingField != null && existingField === name
-    );
-    return newSelectedField;
-  }, [field.value, indices]);
+  const foundField = field.value && indices.fields.find(({ name }) => name === field.value);
 
-  const fieldTypeFilter = useMemo(() => [fieldType], [fieldType]);
+  const { selectedField, indexPattern } = useMemo(() => {
+    /* eslint-disable @typescript-eslint/no-shadow */
+    /* If the field is not found in the indices, we still want to show it in the dropdown */
+    const selectedField = foundField || { name: field.value, type: fieldType };
+    const indexPattern = foundField
+      ? indices
+      : { ...indices, fields: [...indices.fields, selectedField] };
+
+    return { selectedField, indexPattern };
+    /* eslint-enable @typescript-eslint/no-shadow */
+  }, [field.value, indices, foundField, fieldType]);
```

Comment:
For better readability this logic could be moved to an "internal" hook named like `useSelectedField()` and placed under the component in the file.

Though it might be better to have this logic in `FieldComponent`.